### ISSUE TITLE
Short and/or long k-mer types.

### ISSUE THREAD

**Issue by bicycle1885** [BioJulia/Kmers.jl#1]
Currently, we have a kmer type, `Kmer`, which is a 64-bit primitive. Occasionally we would want shorter and/or longer kmer types for some purpose. So, what about having multiple kmer types like integer types of the standard library? I think it would be sensible to have an alias `Kmer` to the current kmer type (e.g. 64-bit kmer) and provide other kmer types such as `Kmer8`, `Kmer16`, ..., `Kmer128`.

**Comment by Ward9250** [comment]
Yeah my colleague works a lot on k-mers and wants longer k-mers. Another possibility is for the data-field to be parametric for different N's of Ntuples if longer k-mers are required than allowed by 128. I will have to ask how long the kmers used in his assembler and contigers in the morning.

**Comment by Ward9250** [comment]
How would defining separate primitives compare to a simple wrapping parametric type such as below? With separate primitive types we have to do a fair bit of loop with eval-ing, but with a parametric type, quite a lot of this can be simplified by the ability to use `I` as a method parameter. I believe I read somewhere simple wrapping types used for the purposes of dispatch have little downside, and since it's immutable, like primitive types, these will be passed by value and laid out in arrays too.   

```julia
immutable Kmer{T<:NucleicAcid, K, I<:Unsigned} <: Sequence
    data::I
end
```

**Comment by kdmurray91** [comment]
@Ward9250 I like your suggestion of `I <: Unsigned`. The other possibility is a [statically sized array](https://github.com/JuliaArrays/StaticArrays.jl) of UInt. I assume the optimizer would collapse `kmer.data[1]` down to something that doesn't involve the inefficiency of array indexing (especially with `@inbounds`)?

**Comment by Ward9250** [comment]
@kdmurray91 That sounds really cool, we can do a quick implementation and then benchmark it to see.
